tests/phpunit/unit-tests/class-llms-test-engagements.php

```diff
@@ -12,6 +12,81 @@
  */
 class LLMS_Test_Engagements extends LLMS_UnitTestCase {
 
+	/**
+	 * Returns a mock 3rd party engagements class.
+	 *
+	 * @since [version]
+	 *
+	 * @return object
+	 */
+	private function instantiate_mock_engagements() {
+
+		$mock_engagements = new class {
+
+			public $engagement_action = 'llms_mock_curriculum_completed';
+
+			public $event_type = 'diploma';
+
+			public $handler_action = 'lifterlms_engagement_ship_diploma';
+
+			public $post_type = 'llms_mock_diploma';
```

Comment:
I think tests are failing because this post type isn't registered. Register it during setup and deregister during teardown and I think that'll fix the test errors.

Reply:
Dang it! I saw this failure before I pushed the branch and fixed it (by using a known post type) before I opened the PR. Then I found another issue with the test and while fixing that one, re-introduced the first one. I didn't notice it because running PHPUnit on just this file works fine. But it fails when running the full suite.

Thanks for your suggestion. It saved me some time.